lib/kwtool_web/plugs/set_current_user.ex

```diff
@@ -0,0 +1,21 @@
+defmodule KwtoolWeb.Plugs.SetCurrentUser do
+  import Plug.Conn
+
+  alias Kwtool.Accounts.Guardian.Plug
+
+  def init(_options), do: nil
+
+  def call(conn, _params) do
+    current_user = Plug.current_resource(conn)
+
+    if current_user do
+      conn
+      |> assign(:current_user, current_user)
+      |> assign(:user_signed_in?, true)
```

Comment:
Do we need this one? can we check if `current_user` is nil or not? 🤔

Reply:
yes, i updated it.